client/layout/index.jsx

```diff
@@ -204,10 +208,22 @@ const Layout = createReactClass( {
 
 export default connect( state => {
 	const { isLoading, section } = state.ui;
+
+	const recentSitesPreference = getPreference( state, 'recentSites' );
+	const primarySiteId = getPrimarySiteId( state );
+
+	let siteIdToFetch = get( recentSitesPreference, '0', null );
+
+	if ( ! siteIdToFetch ) {
+		siteIdToFetch = primarySiteId;
+	}
+
 	return {
 		isLoading,
-		isSupportUser: state.support.isSupportUser,
 		section,
+		siteIdToFetch,
+		shouldFetchSite: getSitesItems( state ) === EMPTY_SITES,
```

Comment:
There might be a potential bug here which I just realized: let's say you have fresh cache. You go to wp.com and your primary site loads. Fine. Now, the primary site caches itself + `/me/preferences` loads as well and caches itself. You don't clear the cache and go to wp.com again. Now, the most recent site should load according to logic (since preferences are already loaded). However, since `getSitesItems` are not `EMPTY_SITES` any more, it won't fetch the most recent site. The problem also is that it won't even re-fetch the primary site so if there was a change in wp-admin, it won't be shown in Calypso (it might be since we have multiple QuerySites across the codebase but still a bug).